registry/repository.go

```diff
@@ -79,6 +79,9 @@ func (c *repositoryClient) PushSignature(ctx context.Context, blob []byte, media
 	if err != nil {
 		return ocispec.Descriptor{}, ocispec.Descriptor{}, err
 	}
+	if manifestDesc.MediaType != ocispec.MediaTypeArtifactManifest {
+		return ocispec.Descriptor{}, ocispec.Descriptor{}, fmt.Errorf("manifestDesc.MediaType requires %q, got %q", ocispec.MediaTypeArtifactManifest, manifestDesc.MediaType)
+	}
```

Comment:
Is this a sanity check?

Reply:
The `oras.Pack()` always return an `ocispec.MediaTypeArtifactManifest` type by the given options.

Reply:
I see. Then this check is redundant, will remove it.